sdk/compute/Azure.ResourceManager.Compute/tests/Unit/VirtualMachineExtensionDataTests.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using Azure.Core;
+using Azure.ResourceManager.Compute.Models;
+using Azure.ResourceManager.Resources.Models;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Compute.Tests.Unit
+{
+    public class VirtualMachineExtensionDataTests
+    {
+        private const string secretUri = "https://kvname.vault.azure.net/secrets/secretname/79b88b3a6f5440ffb2e73e44a0db712e";
+        private const string vaultId = "/subscriptions/a53f7094-a16c-47af-abe4-b05c05d0d79a/resourceGroups/myResourceGroup/providers/Microsoft.KeyVault/vaults/kvName";
+
+        [TestCase]
+        public void ValidateSetBinaryDataGetFromModel()
```

Comment:
We should have these for all BinaryData to Concrete conversions since it looks like a few of them have no shared memory backer.